csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -401,6 +419,41 @@ void FusionKernelRuntime::compileFusionParallel(KernelArgumentHolder args) {
     num_live_args_after_segment_runs_.push_back((int64_t)args.size());
   }
 
+  // add all expressions and compiled kernels to the host ir container
+  if (isOptionEnabled(EnableOption::HostIrLowering)) {
+    IrCloner ir_cloner(hic.get());
+    FusionGuard::setCurFusion(hic.get());
+    for (int64_t run_order_id = 0; run_order_id < num_groups; ++run_order_id) {
+      auto group_to_run = runtime_workspace_.group_run_order.at(run_order_id);
+      auto in_clone = ir_cloner.clone(group_to_run->inputs());
+      auto out_clone = ir_cloner.clone(group_to_run->outputs());
+      if (hic->hasKernelExecutor(run_order_id)) {
```

Comment:
I believe your test only covers one path (probably the if-else) not the other. Either write more tests or put a `NVF_CHECK` saying not implemented. I prefer the latter: https://google.github.io/eng-practices/review/developer/small-cls.html.

Reply:
My sum test will run the `then` block and the set test will run the `else` block in this if statement.